### PR TITLE
Closes #258 delete plan

### LINKED ISSUE #258: Features:
-admin should have ability to delete plans
-mentor should have ability to delete any plan which is created by him

### PR DESCRIPTION
**Pull request by SofiiaFlys**

### CODE REVIEW

File: src/app/plan/plans/plans.component.ts [hunk 1]
```diff
@@ -9,9 +10,22 @@ import { PlanService } from '../../common/services/plan.service';
...
+    planService.getPlans().subscribe( (x: Plan[]) => this.plans = x);
   }
   ngOnInit() {
+    if (this.authService.isAdmin() || this.authService.isMentor()) {
+      this.hasPermisionsToDelete = true;
+    }
+  }
+
+  onDelete(id: number): void {
+    this.id = id;
```
Comment (YuLad) [inline src/app/plan/plans/plans.component.ts]: Why do you need `this.id`, the argument `id` should be enough.

File: src/app/plan/plans/plans.component.ts [hunk 2]
```diff
@@ -9,9 +10,22 @@ import { PlanService } from '../../common/services/plan.service';
...
+    this.planService.deletePlanById(this.id).subscribe();
+    this.index = this.plans.findIndex((plan: Plan) => plan.Id === id);
```
Comment (YuLad) [inline src/app/plan/plans/plans.component.ts]: Same comment with `this.index`, why store additional unused state, local variable should be enough, e.g. `let index = ...`

File: src/app/plan/plans/plans.component.ts (hunk 1 above)
Comment (SofiiaFlys) [reply src/app/plan/plans/plans.component.ts]: ok, I'll remove it

File: src/app/plan/plans/plans.component.ts (hunk 2 above)
Comment (SofiiaFlys) [reply src/app/plan/plans/plans.component.ts]: ok, i'll change it

**Pull request merged**